x/nexus/keeper/rate_limit.go

```diff
@@ -124,6 +114,22 @@ func (k Keeper) getTransferEpoch(ctx sdk.Context, chain exported.ChainName, asse
 	return transferEpoch, k.getStore(ctx).GetNew(getTransferEpochKey(chain, asset, direction), &transferEpoch)
 }
 
+func (k Keeper) getCurrentTransferEpoch(ctx sdk.Context, chain exported.ChainName, asset string, direction exported.TransferDirection, window time.Duration) types.TransferEpoch {
+	transferEpoch := funcs.MustOk(k.getTransferEpoch(ctx, chain, asset, direction))
+
+	// use a new transfer epoch if there was none or if the epoch is outdated
+	epoch := computeEpoch(ctx, window)
+	if transferEpoch.Epoch != epoch {
+		return types.NewTransferEpoch(chain, asset, epoch, direction)
+	}
+
+	return transferEpoch
```

Comment:
the panic seems dangerous
```suggestion
	// use a new transfer epoch if there was none or if the epoch is outdated
	epoch := computeEpoch(ctx, window)
	if transferEpoch, found := k.getTransferEpoch(ctx, chain, asset, direction); found && transferEpoch.Epoch == epoch {
		return transferEpoch
	}
	
	return types.NewTransferEpoch(chain, asset, epoch, direction)
	
```

Reply:
a transfer epoch should always be present if a rate limit is set, but i'm comfortable with not enforcing that in this function